### PR TITLE
added validation on tagname

### LINKED ISSUE #261: Improve checking of tag names
Currently we delegate to the git command line for checking whether a tag name is allowed. We could do more checking in Django to give a friendly (& earlier) error. This would also let us give a much more tightly defined URL regex for entity versions. For instance, we could say tag names must be drawn from `[-_A-Za-z0-9]`. This is more constrained than [git allows](https://git-scm.com/docs/git-check-ref-format) but probably covers what users would actually want.

Then we could set tag form fields to validate against this regex too.

Follows from #260.

### PR DESCRIPTION
**Pull request by sroderick-g5sro**
Fixes #261.

### CODE REVIEW

File: weblab/entities/forms.py [hunk 1]
```diff
@@ -57,7 +58,7 @@ class EntityVersionForm(forms.Form):
     )
     tag = forms.CharField(
         help_text='Optional short label for this version',
-        required=False)
+        required=False, validators=[RegexValidator(r'[-_A-Za-z0-9]', 'Please enter a valid tag name.')])
```
Comment (jonc125) [inline weblab/entities/forms.py:61]: Just a stylistic thing: it's best practice to have each arg on a separate line when they don't all fit on one.
```suggestion
        required=False,
        validators=[RegexValidator(r'[-_A-Za-z0-9]', 'Please enter a valid tag name.')])
```

File: weblab/entities/forms.py [hunk 2]
```diff
@@ -94,7 +95,7 @@ class EntityTagVersionForm(forms.Form):
     tag = forms.CharField(
         label='New tag',
         help_text='Short label for this version',
-        required=True)
+        required=True, validators=[RegexValidator(r'[-_A-Za-z0-9]', 'Please enter a valid tag name.')])
```
Comment (jonc125) [inline weblab/entities/forms.py:98]:
```suggestion
        required=True,
        validators=[RegexValidator(r'[-_A-Za-z0-9]', 'Please enter a valid tag name.')])
```

**Review by jonc125 — CHANGES_REQUESTED** [review]
I can confirm that this works as expected, albeit it doesn't solve #272 or #229 unfortunately. It would be good if the error message said what *was* valid, e.g. add `(only letters, numbers, dashes or underscores are allowed)`.

File: weblab/entities/forms.py (hunk 1 above)
Comment (sroderick-g5sro) [reply weblab/entities/forms.py:61]: done

Comment (jonc125) [reply weblab/entities/forms.py:61]: The other occurrence is done, but you missed this one :)

Comment (sroderick-g5sro) [reply weblab/entities/forms.py]: sorry, changed that now

**Review by jonc125 — APPROVED** [review]
Great!

**Pull request merged**